benchmarks/linear_algebra/blas/level3/symm/symm_generator.cpp

```diff
@@ -30,64 +32,58 @@ using namespace tiramisu;
 
-	computation SIZES("{SIZES[e]: 0<=e<2}", expr(), false, p_float64, &symm);
-
 	// Constants
-	constant N("N", SIZES(0), p_int32, true, NULL, 0, &symm);
-	constant M("M", SIZES(1), p_int32, true, NULL, 0, &symm);
+	constant N("N", expr((int32_t) N)); 
+	constant M("M", expr((int32_t) M))
 
 	// Iteration variables
-	var i("i"), j("j"), k("k");
+	var i("i", 0, N), j("j", 0, M), k("k", 0, j);
 
 	// Scalars  
-	tiramisu::computation alpha("{alpha[0]}", expr(), false, p_float64, &symm);
-	tiramisu::computation beta("{beta[0]}", expr(), false, p_float64, &symm);
-	tiramisu::constant a("a", alpha(0),p_float64, true, NULL, 0, &symm);
-	tiramisu::constant b("b", beta(0), p_float64,true, NULL, 0,   &symm);
+	computation alpha("{alpha[0]}", expr());
+	computation beta("{beta[0]}", expr());
+	constant a("a", alpha(0));
+	constant b("b", beta(0));
 
 	// Matrix 
-	computation B("[N,M]->{B[k,j]: 0<=k<N and 0<=j<M}", expr(), false, p_float64, &symm);
-	computation A("[N]->{A[i,k]: 0<=i<N and 0<=k<N}", expr(), false, p_float64, &symm);
+	input B("B", {"k", "j"}, {N, M}, p_float64);
+	input A("A", {"i", "k"}, {N, N}, p_float64);
     	
-	computation C_0("[N,M]->{C_0[i,j]: 0<=i<N and 0<=j<M}", expr(), false, p_float64, &symm);
+	input C_0("C_0", {"i", "j"}, {N, M}, p_float64);
 	
 	// Computations 
 	// Initialisation à 0 
-	computation init("[N, M]->{init[i,j]: 0<=i<N and 0<=j<=M}", expr(cast(p_float64, 0)), true, p_float64, &symm);
+	computation init("init", {i,j}, expr((p_float64) 0));
 	
 	//  A * B
-	computation mat_mul_a_b("[N, M]->{mat_mul_a_b[i,j,k]: 0<=i<N and 0<=j<=M and 0<=k<j-1}", expr(), true, p_float64, &symm);
+	computation mat_mul_a_b("mat_mul_a_b", {i,j,k}, p_float64);
 
 	// a * temp 
-	computation mult_alpha("[N, M]->{mult_alpha[i,j,k]: 0<=i<N and 0<=j<=M and 0<=k<j-1}", expr(), true, p_float64, &symm);
+	computation mult_alpha("mult_alpha",  {i,j,k}, p_float64);
 
 	// Reduction on C 
-	computation C_1("[N,M]->{C_1[i,j,k]: 0<=k<N and 0<=j<M}", C_0(k-1,j) + mult_alpha(i,j,k) , false, p_float64, &symm);
+	computation C_1("C_1", {i,j,k}, C_0(k-1,j) + mult_alpha(i,j,k));
 
 
 	//  B * A 
-	computation mat_mul_b_a("[N, M]->{mat_mul_b_a[i,j,k]: 0<=i<N and 0<=j<=M and 0<=k<j-1}", expr(), true, p_float64, &symm);
-
+	computation mat_mul_b_a("mat_mul_b_a", {i,j,k}, p_float64);
 
-	computation mat_mul_a_b_e("[N, M]->{mat_mul_a_b_e[i,j]: 0<=i<N and 0<=j<=M }", expr(), true, p_float64, &symm);
 
-	
-	computation mult_alpha_k("[N, M]->{mult_alpha_k[i,j]: 0<=i<N and 0<=j<=M}", expr(), true, p_float64, &symm);
+	computation mat_mul_a_b_e("mat_mul_a_b_e", {i,j}, p_float64);
 
-	computation mult_beta("[N, M]->{mult_beta[i,j]: 0<=i<N and 0<=j<M}", expr(), true, p_float64, &symm);
 	
+	computation mult_alpha_k("mult_alpha_k", {i,j}, p_float64);
 
-	computation add_all("[N, M]->{add_all[i,j]: 0<=i<N and 0<=j<=M }", expr(), true, p_float64, &symm);
+	computation mult_beta("mult_beta", {i,j}, p_float64);	
```

Comment:
Please remove the spaces at the end. These spaces do not respect the Tiramisu coding style.

Reply:
Removed.